### PR TITLE
nf-test for ashlar

### PR DESCRIPTION
**Pull request by kbestak**
Migrate pytest to nf-test for ASHLAR

## PR checklist

Closes https://github.com/nf-core/modules/issues/6895

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda` --- CONDA fails tests - not reproducible between Docker - Conda?

### CODE REVIEW

File: modules/nf-core/ashlar/main.nf [hunk 1]
```diff
@@ -49,4 +49,25 @@ process ASHLAR {
         ashlar: \$(ashlar --version | sed 's/^.*ashlar //' )
     END_VERSIONS
     """
+
+    stub:
+    def args          = task.ext.args           ?: ''
```
Comment (SPPearce) [inline modules/nf-core/ashlar/main.nf:54]:
```suggestion
```

**Review by SPPearce — APPROVED** [review]

**Comment by kbestak** [comment]
Thanks for the review @SPPearce, I don't fully understand why the conda test is failing as the tool is on bioconda and deployed as a biocontainer. I would assume that's an issue. Should the conda test be skipped?

**Comment by SPPearce** [comment]
There is presumably something inside the test_all.ome.tif file that is different between docker/singularity and conda.
This happens often for some file types, such as bam files, where the header of the file encodes a file path that is different between the two environments (i.e. `/opt/bin` instead of /`usr/bin` or something).
Generally at that point we test for existence of the file, or the presence of a string rather than the entire file.

**Comment by kbestak** [comment]
The issue I'm seeing is that the conda output matched the md5sum in the previous test, even though the environment definition was the same.

**Comment by kbestak** [comment]
I'm also noticing an issue with application of flat and dark field correction - should be looked into before merging.

**Comment by kbestak** [comment]
A short update, I've just tested and a newly built Seqera container with ASHLAR from bioconda produces the same outputs as the conda version, but I haven't yet found the precise reason for the difference.

**Pull request merged**